src/org/scada_lts/web/mvc/api/MailingListAPI.java

```diff
@@ -75,4 +106,96 @@
         }
     }
 
+    @GetMapping(value = "/generateUniqueXid")
+    public ResponseEntity<String> generateUniqueXid(HttpServletRequest request) {
+        LOG.info("/api/mailingList/generateUniqueXid");
+        try {
+            User user = Common.getUser(request);
+            if (user != null) {
+                return new ResponseEntity<>(mailingListService.generateUniqueXid(), HttpStatus.OK);
+            } else {
+                return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+            }
+        } catch (Exception e) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @PostMapping(value = "", produces = "application/json")
+    public ResponseEntity<String> createMailingList(@RequestBody MailingList mailingList, HttpServletRequest request) {
+        LOG.info("POST:/api/mailingList");
+        try {
+            User user = Common.getUser(request);
+            if (user != null && user.isAdmin()) {
+                String error = validateMailingListCreate(mailingList);
+                if (!error.isEmpty()) {
+                    return ResponseEntity.badRequest().body(formatErrorsJson(error));
+                }
+                updateValueMailingListPost(mailingList);
+                mailingListService.saveMailingList(mailingList);
+                return new ResponseEntity<>("{\"status\":\"created\"}", HttpStatus.CREATED);
+            } else {
+                return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+            }
+        } catch (Exception e) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @PutMapping(value = "", produces = "application/json")
+    public ResponseEntity<String> updateMailingList(@RequestBody MailingListDTO mailingList, HttpServletRequest request) {
```

Comment:
I propose to prepare 4 structures, one for creating an object, CreateMailinglist (request, without id and xid) and CreatedMailinglist (response with generated id and xid), it will also simplify validation, xid should be generated on the java side, id is generated by the database. The second structure UpdateMailinglist (request) and UpdatedMailinglist (response), with id and xid, then I suggest creating a method for get the Mailinglist, if it does not find the object in the database after id/xid, it would be good to return 404. 

If the main task of an API method is to update object and this object does not exist in the database, then returned should be 404, not 400. In validation, I would try to limit the situations in which a database operation has to be performed.